File: ZeroSupport/app/src/main/java/com/risakokato/zerosupport/list/BelongingsAdapter.java
package com.risakokato.zerosupport.list;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.risakokato.zerosupport.model.Belongings;
import com.risakokato.zerosupport.R;

import java.util.List;

public class BelongingsAdapter extends ArrayAdapter<Belongings> {

    private LayoutInflater layoutinflater;

    public BelongingsAdapter(Context context, int textViewResourceId, List<Belongings> objects) {
        super(context, textViewResourceId, objects);
        layoutinflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        Belongings belo= getItem(position);

        if (convertView == null) {
            convertView = layoutinflater.inflate(R.layout.layout_item_list, null);

        }
        TextView dateText = (TextView) convertView.findViewById(R.id.dateText);
        TextView belongingText = (TextView) convertView.findViewById(R.id.belongingsText);

        dateText.setText(belo.date);
        belongingText.setText(belo.belongings);

        return convertView;

    }
}

File: ZeroSupport/app/src/main/java/com/risakokato/zerosupport/list/BelongingsAdapter.kt
package com.risakokato.zerosupport.list

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.recyclerview.widget.RecyclerView
import com.risakokato.zerosupport.R
import com.risakokato.zerosupport.model.entity.BelongingsRoom

class BelongingsAdapter(private val context: Context?, private var mList: List<BelongingsRoom>) : RecyclerView.Adapter<BelongingsAdapter.ViewHolder>() {

    lateinit var listener: OnClickListener

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val vh = ViewHolder(LayoutInflater.from(context).inflate(R.layout.layout_item_list, parent, false))
        return vh
    }

    override fun getItemCount(): Int = mList.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.belongings.text = mList[position].belongings
        holder.dateText.text = mList[position].date
        holder.view.setOnLongClickListener {
            listener.setOnLongClickListener(position)
            true
        }
    }

    fun setList(new: List<BelongingsRoom>) {
        mList = new
    }

    interface OnClickListener {
        fun setOnLongClickListener(position: Int)
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val view = itemView.findViewById<ConstraintLayout>(R.id.container)
        val belongings = itemView.findViewById<TextView>(R.id.belongingsText)
        val dateText = itemView.findViewById<TextView>(R.id.dateText)
    }

}